addons/l10n_account_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,350 @@
+            if not withholding_line.name:
+                withholding_line.name = withholding_line.tax_id.l10n_account_wth_sequence_id.next_by_id()
+
+            # We need to support using multiple tax repartition lines, but in the wizard one tax = one line.
+            # So, we'll only split the result here. If the amount of the line has been set manually, _get_withholding_tax_values already returns the tax details for that amount.
+            for tax in withholding_tax_values:
+                tax_account = tax['account']
+                if not tax_account:
+                    raise UserError(self.env._('Please define a tax account on the distribution of the tax %(tax_name)s', tax_name=tax['tax_name']))
+
+                withholding_line_vals.append({
+                    'currency_id': self.currency_id.id,
+                    'name': f'WH Tax: {withholding_line.name}',
+                    'account_id': tax_account,
+                    'amount_currency': sign * tax['amount'],
+                    'balance': sign * self.currency_id._convert(tax['amount'], self.company_id.currency_id, self.company_id, self.payment_date),
+                    'tax_base_amount': tax['tax_base_amount'],
+                    'tax_repartition_line_id': tax['tax_repartition_line'],
+                    'tax_tag_ids': [Command.set(tax['tag_ids'])],
+                    'analytic_distribution': withholding_line.analytic_distribution,
+                })
+
+        # We also need tax base lines for reporting.
+        # Group the withholding lines by their base amount and base tags.
+        group = self.l10n_account_wth_line_ids.grouped(lambda m: (m.base_amount, self.env['account.account.tag'].browse(m._get_withholding_tax_base_tag_ids())))
+        base_lines_to_create = []
+        # This looping loop aim is to optimize the amount of account.move.line being created by this flow by grouping them.
+        for (group_base_amount, group_tags), withholding_lines in group.items():
+            # This base amount grouped above need to be multiplied in order to correctly affect the tax tags
+            group_base_amount *= len(withholding_lines)
+
+            # If we have a line in our list which has a base amount matching the one of the current group and which also
+            # doesn't have any of the tags inside, we can merge this group with that one to save records.
+            added_to_existing = False
+            for i in range(len(base_lines_to_create)):
+                ((base_amount, tags), lines) = base_lines_to_create[i]
+                if base_amount != group_base_amount or any(group_tag in tags for group_tag in group_tags):
+                    continue
+
+                base_lines_to_create[i] = ((base_amount, tags + group_tags), lines + withholding_lines)
+                added_to_existing = True
+                break
+
+            if not added_to_existing:
```

Comment:
Note that you don't need the `added_to_existing` variable
```suggestion
                break
            else:
```

But it would be more logical by inverting the `if` condition as well
```python
for i in range(len(base_lines_to_create)):
    ((base_amount, tags), lines) = base_lines_to_create[i]
    if (base_amount == group_base_amount) and not (tags & group_tags):
        base_lines_to_create[i] = ((base_amount, tags + group_tags), lines + withholding_lines)
        break
else:
    base_lines_to_create.append(((group_base_amount, group_tags), withholding_lines))
```